storybook/stories/core/components/Input/RadioPillGroup.stories.js

```diff
@@ -0,0 +1,295 @@
+  {
+    id: 4,
+    label: 'Start and End Day',
+  },
+]
+
+const optionsWithCustomStyle = [
+  {
+    id: 1,
+    label: 'All Days',
+  },
+  {
+    id: 2,
+    label: 'Start Day Only',
+    style: { 'background-color': 'lightblue' }
+  },
+  {
+    id: 3,
+    label: 'End Day Only',
+    style: { 'background-color': 'lightgreen' }
+  },
+  {
+    id: 4,
+    label: 'Start and End Day',
+    style: { 'background-color': 'lightyellow' }
+  },
+]
+
+const Template = (args) => ({
+  setup () {
+    const selected = ref(1)
+    return { args, selected }
+  },
+  render () {
+    return h('div', {}, [
+      h(RadioPillGroup, {
+        ...this.args,
+        modelValue: this.selected,
+        'onUpdate:modelValue': (value) => {
+          this.selected = value
+        },
+      }),
+      h('br'),
+      h('p', {}, `v-model : ${this.selected}`),
+    ])
+  },
+})
+
+export const Default = Template.bind({})
+Default.args = {
+  options: options,
+  name: 'partialDays',
+}
+
+Default.parameters = {
+  docs: {
+    source: {
+      code: `<oxd-radio-pill-group
+    name="partialDays"
+    :options=${JSON.stringify(options)} 
+/>`,
+    },
+  },
+}
+
+export const Disabled = Template.bind({})
+Disabled.args = {
+  options: options,
+  disabled: true,
+  name: 'partialDays',
+}
+Disabled.parameters = {
+  docs: {
+    source: {
+      code: `<oxd-radio-pill-group
+    name='partialDays'
+    :options=${JSON.stringify(options)} 
+    :disabled='true'
+/>`,
+    },
+  },
+}
+
+export const ReadOnly = Template.bind({})
+ReadOnly.args = {
+  options: options,
+  readonly: true,
+  name: 'readonlyGroup',
+}
+ReadOnly.parameters = {
+  docs: {
+    source: {
+      code: `<oxd-radio-pill-group
+    name='readonlyGroup'
+    :options=${JSON.stringify(options)} 
+    :readonly='true'
+/>`,
+    },
+  },
+}
+
+const customStyle = { 'padding': '1rem', 'border': '1px solid', 'border-radius': '0.65rem' }
+export const CustomStyles = Template.bind({})
+CustomStyles.args = {
+  options: optionsWithCustomStyle,
+  style: customStyle,
+  name: 'customStyle',
+}
+
+CustomStyles.parameters = {
+  docs: {
+    source: {
+      code: `<oxd-radio-pill-group
+    name='customStyle'
+    :options=${JSON.stringify(optionsWithCustomStyle)}
+    :style=${JSON.stringify(customStyle)}
+/>`,
+    },
+  },
+}
+
+const EventsTemplate = (args) => ({
+  setup () {
+    const groupValue = ref(1)
+    return {
+      args,
+      groupValue
+    }
+  },
+  template: `
+    <oxd-radio-pill-group
+        name="events"
+        v-bind="args"
+        v-model="groupValue"
+        @update:modelValue="updateModelValue"
+        @click="click"
+        @focus="focus"
+        @change="change"
+        @blur="blur"/>`,
+
+  components: { 'oxd-radio-pill-group': RadioPillGroup },
+  methods: actions({
+    focus: 'focus',
+    blur: 'blur',
+    change: 'change',
+    'updateModelValue': 'update:modelValue',
+    click: 'click'
+  }),
+})
+
+export const Events = EventsTemplate.bind({})
```

Comment:
When triggering the event's better to indicate it in the frontend. Check other widgets done by Thilina

Reply:
Events are logged to Actions tab in storybook